group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -593,6 +609,9 @@ private void throwIfConsumerGroupHeartbeatRequestIsInvalid(
             if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
                 throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
             }
+        } else if (request.memberEpoch() == -2) {
+            throwIfEmptyString(request.memberId(), "MemberId can't be empty.");
```

Comment:
This is for the case where the static member is leaving temporarily, right?

Would it be possible to add more detail to these error messages to aid in troubleshooting/debugging on the client when this condition is hit?

Reply:
> This is for the case where the static member is leaving temporarily, right

Yes, that's correct. I have added some debugging information (like groupId etc). Let me know if that makes sense.

Reply:
Also, `MemberId can't be empty.` string is used in other places as well (when member epoch is > 0  or equal to -1. Should we look to change those as well?

Reply:
It's just my preference, so if there's precedent for how you have it, I wouldn't hold up this PR in an effort to change the other places. Thanks!

Reply:
Makes sense. I think I updated that one comment. But yeah the rest of the loggers, I won't be touching them, as that would be noisy to review.